Classes/ContextMenu/ContentAiTranslationProvider.php

```diff
@@ -0,0 +1,237 @@
+
+class ContentAiTranslationProvider extends AbstractProvider
+{
+    /**
+     * @var array<string, array{
+     *     type: string,
+     *     label: string,
+     *     iconIdentifier: string,
+     *     callbackAction: string
+     * }>
+     */
+    protected $itemsConfiguration = [
+        'translateContentEasy' => [
+            'type' => 'item',
+            'label' => 'LLL:EXT:mkcontentai/Resources/Private/Language/locallang_contentai.xlf:labelTextTranslateContentEasy',
+            'iconIdentifier' => 'actions-edit-copy',
+            'callbackAction' => 'translateContentEasy',
+        ],
+        'translateContentPlain' => [
+            'type' => 'item',
+            'label' => 'LLL:EXT:mkcontentai/Resources/Private/Language/locallang_contentai.xlf:labelTextTranslateContentPlain',
+            'iconIdentifier' => 'actions-edit-copy',
+            'callbackAction' => 'translateContentPlain',
+        ],
+    ];
+
+    public function setContext(string $table, string $identifier, string $context = ''): void
+    {
+        $this->table = $table;
+        $this->identifier = $identifier;
+        $this->context = $context;
+    }
+
+    public function canHandle(): bool
+    {
+        return 'tt_content' === $this->table;
+    }
+
+    public function getPriority(): int
+    {
+        return 55;
+    }
+
+    /**
+     * @param array<string, array{
+     *     type: string,
+     *     label: string,
+     *     iconIdentifier: string,
+     *     additionalAttributes: array<string,string>,
+     *     callbackAction: string
+     * }> $items
+     *
+     * @return array<string, array{
+     * type: string,
+     * label: string,
+     * iconIdentifier: string,
+     * additionalAttributes: array<string,string>,
+     * callbackAction: string
+     * }>
+     */
+    public function addItems(array $items): array
+    {
+        $this->initDisabledItems();
+        $localItems = $this->prepareItems($this->itemsConfiguration);
+
+        return $items + $localItems;
+    }
+
+    public function getItemsConfiguration(): array
+    {
+        return $this->itemsConfiguration;
+    }
+
+    /**
+     * This method is called for each item this provider adds and checks if given item can be added.
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function canRender(string $itemName, string $type): bool
+    {
+        $canRender = false;
+        $availableActions = ['translateContentEasy', 'translateContentPlain'];
+
+        if (in_array($itemName, $availableActions)) {
+            $canRender = $this->isPageContent() && $this->isValidTypeOfRecord((int) $this->identifier);
+        }
+
+        return $canRender;
+    }
+
+    public function isPageContent(): bool
+    {
+        return 'tt_content' === $this->table;
+    }
+
+    public function generateUrl(string $itemName): UriInterface
+    {
+        $typo3Version = GeneralUtility::makeInstance(Typo3Version::class);
```

Comment:
No need for `GeneralUtility::makeInstance()` here, because this class can be considered final:

```suggestion
        $typo3Version = new Typo3Version();
```